src/transformStateWithClones.js

```diff
@@ -7,7 +7,51 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const results = [];
+
+  for (const action of actions) {
+    const newRecord = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties':
+        addData(newRecord, action.extraData);
```

Comment:
The `addData` function mutates the `newRecord` object directly. Consider creating a new object with the added properties instead of mutating the existing one.